src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5391,6 +5391,15 @@ private BoundExpression MakeBadExpressionForObjectCreation(SyntaxNode node, Type
                         break;
                     }
 
+                case BoundKind.ImplicitIndexerAccess:
+                    var implicitIndexer = (BoundImplicitIndexerAccess)boundMember;
+                    MessageID.IDS_ImplicitIndexInitializer.CheckFeatureAvailability(diagnostics, implicitIndexer.Syntax);
+
+                    hasErrors |= isRhsNestedInitializer
+                        && !CheckNestedObjectInitializerPropertySymbol(GetPropertySymbol(implicitIndexer, out _, out _), leftSyntax, diagnostics, hasErrors, ref resultKind);
+
+                    return hasErrors ? boundMember : CheckValue(boundMember, valueKind, diagnostics);
```

Comment:
>CheckValue(boundMember, valueKind, diagnostics) [](http://example.com/codeflow?start=53&length=47)

It is not obvious why this is necessary and what effect is this going to have. For example, it looks like we are not doing this  for a regular indexer.

Reply:
The regular indexer case (`BoundKind.IndexerAccess`) does this after the `switch` (line 5429 below). 
The effect is shown in tests like `InObjectInitializer_Readonly` and `InObjectInitializer_Readonly`.